saint-app/domain/src/interactors/saint_mutation.rs

```diff
@@ -1,16 +1,49 @@
 use crate::boundaries;
-use crate::boundaries::{SaintDbGateway, SaintMutationRequest, SaintMutationResponse};
+use crate::boundaries::{
+    SaintDbGateway, SaintDbRequest, SaintMutationRequest, SaintMutationResponse,
+};
 use async_trait::async_trait;
 use futures::executor::block_on;
+use uuid::Uuid;
 
 pub struct SaintMutationInteractor {
     db_gateway: Box<dyn SaintDbGateway>,
 }
 
 impl boundaries::SaintMutationInputBoundary for SaintMutationInteractor {
-    fn create_saint(&self, request: SaintMutationRequest) -> SaintMutationResponse {
-        println!("saint mutation input boundary {}", request.id);
-        SaintMutationResponse {}
+    fn create_saint(&self, request: SaintMutationRequest) -> Option<SaintMutationResponse> {
+        println!("saint mutation input boundary {}", request.id.unwrap());
+        let mut id: Uuid;
+        loop {
+            id = Uuid::new_v4();
+            if block_on((*self).db_gateway.exists_by_id(id.clone())) {
+                println!("this id already exists, continue generate");
+            } else {
+                break;
+            }
+        }
```

Comment:
while loop is dangerous, let's have for 0 <= i <= 5 instead with 500 millisecond of sleep between each iteration. If after the loop, still no id, return error